strawberryfields/program.py

```diff
@@ -202,6 +111,9 @@ class Program:
             ops.BSgate(0.43, 0.1) | (q[1], q[2])
             ops.MeasureFock() | q
 
+    The program objects keep track of the state of the quantum register they act on, using a dictionary of :class:`RegRef` objects.
```

Comment:
Since this is very user facing, I would suggest removing this line (quite advanced, and might confuse if they click on `RegRef`).

Reply:
OK, but wasn't the new convention that the basic user-facing docs are rst files in the `doc` directory (e.g. `doc/introduction/*.rst`), and the docstrings in the Python files are for the API docs for devs and more advanced users? What's the correct place for the slightly lower-level explanations on how the code works?

Reply:
I think the best place are comments in the source code itself